src/v/features/feature_table.cc

```diff
@@ -686,10 +715,35 @@ void feature_table::set_license(security::license license) {
     _license = std::move(license);
 }
 
-void feature_table::revoke_license() { _license = std::nullopt; }
+void feature_table::set_builtin_trial_license(
+  model::timestamp cluster_creation_timestamp) {
+    if (ss::this_shard_id() == 0) {
+        vlog(featureslog.debug, "Initializing builtin trial license.");
+    }
+    _builtin_trial_license_initialized = true;
+    _builtin_trial_license = make_builtin_trial_license(
+      model::to_time_point(cluster_creation_timestamp));
+}
+
+void feature_table::revoke_license() {
+    _license = std::nullopt;
+    _builtin_trial_license = std::nullopt;
+}
 
 const std::optional<security::license>& feature_table::get_license() const {
-    return _license;
+    return _license ? _license : _builtin_trial_license;
+}
+
+bool feature_table::should_sanction() const {
+    if (_license) {
+        return _license->is_expired();
+    } else if (_builtin_trial_license) {
+        return _builtin_trial_license->is_expired();
+    }
+
+    // While we are yet to initialize _builtin_trial_license on cluster
+    // creation, be permissive
+    return _builtin_trial_license_initialized;
```

Comment:
```suggestion
    auto license = get_license();
    if (!license) {
        // ...
        return _builtin_trial_license_initialized;
    } 
    return license->is_expired();    
```
are these 2 identical? Or am i missing an edge case?

Reply:
Yes, they are. I'm happy to switch to that if you think that's simpler.